src/app/SimpleSubscriptionResumptionStorage.cpp

```diff
@@ -0,0 +1,589 @@
+    {                                                                                                                              \
+        auto __err = (expr);                                                                                                       \
+        if (!::chip::ChipError::IsSuccess(__err))                                                                                  \
+        {                                                                                                                          \
+            mStorage->SyncDeleteKeyValue(GetStorageKey(node).KeyName());                                                           \
+            ChipLogError(DataManagement, "%s at %s:%d", ErrorStr(__err), __FILE__, __LINE__);                                      \
+            return __err;                                                                                                          \
+        }                                                                                                                          \
+    } while (false)
+
+CHIP_ERROR SimpleSubscriptionResumptionStorage::FindByScopedNodeId(ScopedNodeId node, SubscriptionList & subscriptions)
+{
+    return FindByScopedNodeId(node, subscriptions, 0);
+}
+CHIP_ERROR SimpleSubscriptionResumptionStorage::FindByScopedNodeId(ScopedNodeId node, SubscriptionList & subscriptions,
+                                                                   size_t allocateExtraSpace)
+{
+    Platform::ScopedMemoryBuffer<uint8_t> backingBuffer;
+    backingBuffer.Calloc(MaxStateSize());
+    if (backingBuffer.Get() == nullptr)
+    {
+        mStorage->SyncDeleteKeyValue(DefaultStorageKeyAllocator::SubscriptionResumptionIndex().KeyName());
+        return CHIP_ERROR_NO_MEMORY;
+    }
+
+    uint16_t len = static_cast<uint16_t>(MaxStateSize());
+
+    if (mStorage->SyncGetKeyValue(GetStorageKey(node).KeyName(), backingBuffer.Get(), len) != CHIP_NO_ERROR)
+    {
+        subscriptions.mSize = 0;
+        mStorage->SyncDeleteKeyValue(GetStorageKey(node).KeyName());
+        return CHIP_NO_ERROR;
+    }
+
+    TLV::ScopedBufferTLVReader reader(std::move(backingBuffer), len);
+
+    DeleteSubscriptionsAndReturnLogErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::AnonymousTag()));
+    TLV::TLVType arrayType;
+    DeleteSubscriptionsAndReturnLogErrorOnFailure(reader.EnterContainer(arrayType));
+
+    size_t subscriptionCount;
+    DeleteSubscriptionsAndReturnLogErrorOnFailure(reader.CountRemainingInContainer(&subscriptionCount));
+    if (subscriptionCount >= CHIP_IM_MAX_NUM_SUBSCRIPTIONS)
+    {
+        mStorage->SyncDeleteKeyValue(GetStorageKey(node).KeyName());
+        return CHIP_ERROR_NO_MEMORY;
+    }
+
+    subscriptions.mSubscriptions =
+        std::unique_ptr<SubscriptionInfo[]>(new (std::nothrow) SubscriptionInfo[subscriptionCount + allocateExtraSpace]);
```

Comment:
We should use the library wrapper so that we are using Platform::New and Platform::Delete:

https://github.com/project-chip/connectedhomeip/blob/9fbcf1cef8c4a42761a7fc23588ca71d75904ff5/src/lib/support/CHIPMem.h#L179

Reply:
I've since revamped the storage API according to Tennessee's iterator suggestion, and so this is no longer there.